src/pool/srv_pool.c

```diff
@@ -2461,13 +2468,25 @@ ds_pool_update(uuid_t pool_uuid, crt_opcode_t opc,
 		} else { /* enabled by default */
 			D_ASSERT(replicas != NULL);
 			ret = ds_rebuild_schedule(pool_uuid, *map_version,
-						  &target_list, replicas);
+						  &target_list, NULL, replicas);
 			if (ret != 0) {
 				D_ERROR("rebuild fails rc %d\n", ret);
 				if (rc == 0)
 					rc = ret;
 			}
 		}
+	} else if (updated && opc == POOL_ADD) {
+		int	 ret;
+
+		/* TODO: Environment variable / config setting to disable add */
+		D_ASSERT(replicas != NULL);
+		ret = ds_rebuild_schedule(pool_uuid, *map_version, NULL,
+					  &target_list, replicas);
```

Comment:
I have been thinking a lot about this - the two options I see are below. I believe the best way to do this is to have separate lists (as in this pull request, option 2 below). 
 
**One list of targets**
Pros: 
- Smaller impact to existing rebuild functions
- Less code duplication
Cons: 
- Each list entry needs to have a way to check if it's being added / removed / etc. We could put this information in each list entry (create a new structure used for list members) or we could infer the operation from the status (assume DOWN = failed/rebuilding, UP = added/reintegrating).
- Some operations will get more complicated - right now there's an API for "set all things in this list to UP_IN" - will have to iterate the list and do these one-at-a-time instead. 

**Multiple lists of targets**
I believe only two lists are needed - a failed list and an added list. Failed will cover both failure and drain use cases, and added will cover both add and reintegration use cases (since these result in all targets being set to DOWNOUT / UPIN respectively upon completion).
Pros: 
- Lists remain "dumb lists" of targets (re-using existing data structure)
- Can operate on entire lists as needed
Cons:  
- Additional arguments to keep track of
- Some code duplication when both failed / added lists require similar operations

**Common**
Regardless of 1 or 2, both approaches will share the same rebuild leader ULT, and share the same scan process. When additional failures / additions / etc. happen, they are merged into the existing list(s) and the scan restarts. 

@gnailzenh Would like to hear your thoughts on this